libs/ts-rest/core/src/lib/client.ts

```diff
@@ -225,3 +239,13 @@ export const initClient = <T extends AppRouter>(
 
   return proxy as InitClientReturn<T>;
 };
+
+// takes a router and returns response types for each AppRoute
+// does not support nested routers, yet
+
+export function getRouteResponses<T extends AppRouter>(router: T) {
+  return {} as {
+     [K in keyof typeof router]: 
+        typeof router[K] extends AppRoute ? ApiResponseForRoute<typeof router[K]> : 'not a route'
+   }
+}
```

Comment:
Not sure how this is testable, tbh. It's a TS feature, not a runtime feature.

Reply:
I also tried a recursive implementation, but it broke the type-checking. Happy to see if you have any ideas

Reply:
Yeah, I'll pull and try to fix the recursive issue!

Reply:
I never had time to tackle this, but I don't want to block this release, I'll open an issue :)